core/include/sched.h

```diff
@@ -148,6 +148,7 @@ typedef struct _thread thread_t;
  */
 typedef enum {
     STATUS_STOPPED,                 /**< has terminated                           */
+    STATUS_CRASHED,                 /**< thread has crashed with a fault          */
```

Comment:
You will also need to patch OpenOCD if you modify this

Reply:
Yup, I'm aware. Do you happen to know how they accept patches?

Reply:
They use [Gerrit](http://openocd.zylin.com/#/q/status:open) - there is a [guide](http://openocd.org/doc-release/doxygen/patchguide.html).

However, you would then have to determine which RIOT version you are on at run-time.
A lot of the hassle could be avoided if you appended the new state at the end of the list.

Reply:
>  A lot of the hassle could be avoided if you appended the new state at the end of the list.

That would require rewriting the [`sched_set_status` function](https://github.com/RIOT-OS/RIOT/blob/master/core/sched.c#L226)

Reply:
You could still make it in time for the next OpenOCD release :-)